### PR TITLE
Implement favorite command

### PR DESCRIPTION
**Pull request by Na-Nazhou**
- Currently type `favorite INDEX` / `fav INDEX` will add tag `Favorite` to the bookmark.
- `fav` is a command alias for `favorite`
- The tag can only removed through `edit INDEX t/`

- Tag `Favorite` shows a different color. Please suggest what color to use :D (for now it looks like this)
image

Todo: (in a separate PR)
- Show favorites on Dashboard
- Avoid the usage of string literal ("Favorite"), probably store as a constant in Tag class.
- Update UG

### CODE REVIEW

**Comment by Na-Nazhou** [comment]
Addresses #93

**Comment by Na-Nazhou** [comment]
Another color:
image

File: src/main/java/seedu/mark/logic/commands/FavoriteCommand.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+/**
+ * Favorites a bookmark identified using its displayed index from Mark.
+ */
+public class FavoriteCommand extends Command {
+    public static final String COMMAND_WORD = "favorite";
+    public static final String COMMAND_ALIAS = "fav";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Favorites the bookmark identified by the index number used in the displayed bookmark list.\n"
```
Comment (dorcastan) [inline src/main/java/seedu/mark/logic/commands/FavoriteCommand.java]:
```suggestion
            + ": Adds a 'Favorite' tag to the bookmark identified by the index number used in the displayed "
            + "bookmark list.\n"
```

File: src/main/java/seedu/mark/logic/commands/FavoriteCommand.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_FAVORITE_BOOKMARK_SUCCESS = "Add to my favorites: %1$s";
```
Comment (dorcastan) [inline src/main/java/seedu/mark/logic/commands/FavoriteCommand.java]:
```suggestion
    public static final String MESSAGE_FAVORITE_BOOKMARK_SUCCESS = "Bookmark added to Favorites: %1$s";
```

File: src/main/java/seedu/mark/logic/commands/FavoriteCommand.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    public static final String MESSAGE_FAVORITE_BOOKMARK_DUPLICATE = "This bookmark already exists in your favorites";
```
Comment (dorcastan) [inline src/main/java/seedu/mark/logic/commands/FavoriteCommand.java]: To match the success message
```suggestion
    public static final String MESSAGE_FAVORITE_BOOKMARK_DUPLICATE = "This bookmark already exists in Favorites";
```

File: src/main/java/seedu/mark/model/Mark.java [hunk 4]
```diff
@@ -187,6 +189,19 @@ public void applyAllTaggers() {
         setBookmarks(autotagController.applyTaggersToList(getBookmarkList()));
     }
 
+
+    //// favorite operations
+
+    /**
+     * Favorites the specified bookmark.
```
Comment (dorcastan) [inline src/main/java/seedu/mark/model/Mark.java]:
```suggestion
     * Tags the specified bookmark as a favorite bookmark.
```

File: src/main/java/seedu/mark/model/Model.java [hunk 5]
```diff
@@ -72,6 +72,12 @@
      */
     void deleteBookmark(Bookmark target);
 
+    /**
+     * Favorites the given bookmark.
```
Comment (dorcastan) [inline src/main/java/seedu/mark/model/Model.java]:
```suggestion
     * Tags the given bookmark as a favorite bookmark.
```

File: src/main/java/seedu/mark/model/ModelManager.java [hunk 6]
```diff
@@ -49,17 +49,7 @@ public ModelManager(ReadOnlyMark mark, ReadOnlyUserPrefs userPrefs) {
...
-                FXCollections.observableArrayList(
-                        new OfflineDocument("example doc",
-                                OfflineDocument.OFFLINE_DOC_EXAMPLE).getCollection()), (
-            Paragraph p1, Paragraph p2) -> {
-            ParagraphIdentifier pid1 = p1.getId();
-            ParagraphIdentifier pid2 = p2.getId();
-            return pid1.compareTo(pid2);
-        }
-        );
+        annotatedDocument = FXCollections.observableList(new ArrayList<>());
```
Comment (Na-Nazhou) [inline src/main/java/seedu/mark/model/ModelManager.java]:
```suggestion
       annotatedDocument = new SortedList<Paragraph>(	        annotatedDocument = FXCollections.observableList(new ArrayList<>());
                //TODO: change this to link to proper offline document	
                FXCollections.observableArrayList(	
                        new OfflineDocument("example doc",	
                                OfflineDocument.OFFLINE_DOC_EXAMPLE).getCollection()), (	
            Paragraph p1, Paragraph p2) -> {	
            ParagraphIdentifier pid1 = p1.getId();	
            ParagraphIdentifier pid2 = p2.getId();	
            return pid1.compareTo(pid2);	
        }	
        );
```

**Pull request merged**